### PR TITLE
Gen3 Monthly Release 2021.07 qa-covid19.planx-pla.net 1625845274

### PR DESCRIPTION
**Pull request by themarcelor**
Applying version 2021.07 to qa-covid19.planx-pla.net

### CODE REVIEW

File: qa-covid19.planx-pla.net/manifest.json [hunk 1]
```diff
@@ -4,33 +4,33 @@
     "That's all I have to say"
   ],
   "versions": {
-    "arborist": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/arborist:2021.04",
-    "augur": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/gen3-augur:fix_docker2",
-    "awshelper": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/awshelper:2021.04",
+    "arborist": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/arborist:2021.07",
+    "augur": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/gen3-augur:2021.07",
```
Comment (paulineribeyre) [inline qa-covid19.planx-pla.net/manifest.json:8]: i think we don't even need augur in the manifest... could you check with @thanh-nguyen-dang, and remove it in QA and prod if we don't need it?
if we need it, we need a create a 0.1.0 tag and deploy that. There is no 2021.07.

File: qa-covid19.planx-pla.net/manifest.json [hunk 2]
```diff
@@ -4,33 +4,33 @@
...
-    "indexd": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/indexd:2021.04",
-    "peregrine": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/peregrine:2021.04",
-    "pidgin": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/pidgin:2021.04",
-    "revproxy": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/nginx:2021.04",
-    "sheepdog": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/sheepdog:2021.04",
-    "portal": "quay.io/cdis/data-portal:feat_WLSmapDataOverlays",
-    "tube": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/tube:0.5.0",
+    "covid19-etl": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/covid19-etl:2021.07",
+    "covid19-notebook-etl": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/covid19-notebook-etl:2021.07",
+    "covid19-bayes-model": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/covid19-bayes-model:2021.07",
```
Comment (paulineribeyre) [inline qa-covid19.planx-pla.net/manifest.json:13]: these are not included in the monthly release

File: qa-covid19.planx-pla.net/manifest.json [hunk 3]
```diff
@@ -4,33 +4,33 @@
...
+    "fence": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/fence:2021.07",
+    "indexd": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/indexd:2021.07",
+    "peregrine": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/peregrine:2021.07",
+    "pidgin": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/pidgin:2021.07",
+    "revproxy": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/nginx:2021.07",
+    "sheepdog": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/sheepdog:2021.07",
+    "portal": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/data-portal:2021.07",
```
Comment (paulineribeyre) [inline qa-covid19.planx-pla.net/manifest.json:20]: please check with @ocshawn And @hughestr

File: qa-covid19.planx-pla.net/manifest.json [hunk 4]
```diff
@@ -63,16 +63,16 @@
   },
   "ssjdispatcher": {
     "job_images": {
-      "indexing": "quay.io/cdis/indexs3client:2021.04"
+      "indexing": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/indexs3client:2021.07"
```
Comment (paulineribeyre) [inline qa-covid19.planx-pla.net/manifest.json:66]: are all these job images in ECR?

**Review by paulineribeyre — CHANGES_REQUESTED** [review]

File: qa-covid19.planx-pla.net/manifest.json (hunk 2 above)
Comment (themarcelor) [reply qa-covid19.planx-pla.net/manifest.json:13]: shouldn't they be included?

Comment (paulineribeyre) [reply qa-covid19.planx-pla.net/manifest.json:13]: no, updates to the PRC ETLs are usually small and urgent to deploy, there never are changes on master waiting to be deployed

Comment (atharvar28) [reply qa-covid19.planx-pla.net/manifest.json:13]: I have added them to the ignore list for jenkins job. So next time when the job is run, the job wouldnt replace the version of these services @paulineribeyre

File: qa-covid19.planx-pla.net/manifest.json (hunk 1 above)
Comment (atharvar28) [reply qa-covid19.planx-pla.net/manifest.json:8]: checked with Thanh / Yilin if we need augur in the manifest.json. @yilinxu mentioned it is ok to remove augur from the manifest. I will remove it from PROD manifest when deploying the new versions to it.

File: qa-covid19.planx-pla.net/manifest.json (hunk 4 above)
Comment (atharvar28) [reply qa-covid19.planx-pla.net/manifest.json:66]: Yes all these job images are in ECR. After the release version is released, there is a job in release pipeline which pushes all the quay images to AWS ECR.

File: qa-covid19.planx-pla.net/manifest.json (hunk 3 above)
Comment (atharvar28) [reply qa-covid19.planx-pla.net/manifest.json:20]: Need to revert back to using the branch after testing is done

File: qa-covid19.planx-pla.net/manifest.json (hunk 1 above)
Comment (paulineribeyre) [reply qa-covid19.planx-pla.net/manifest.json:8]: nice, let's remove it here too then 👍

File: qa-covid19.planx-pla.net/manifest.json [hunk 5]
```diff
@@ -4,33 +4,32 @@
...
-    "covid19-bayes-model": "quay.io/cdis/covid19-bayes-model:3.4.3",
-    "fence": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/fence:4.29.1",
-    "indexd": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/indexd:2021.04",
-    "peregrine": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/peregrine:2021.04",
-    "pidgin": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/pidgin:2021.04",
-    "revproxy": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/nginx:2021.04",
-    "sheepdog": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/sheepdog:2021.04",
-    "portal": "quay.io/cdis/data-portal:feat_WLSmapDataOverlays",
-    "tube": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/tube:0.5.0",
+    "covid19-bayes-model": "707767160287.dkr.ecr.us-east-1.amazonaws.com/gen3/covid19-bayes-model:3.4.3",
```
Comment (paulineribeyre) [inline qa-covid19.planx-pla.net/manifest.json:12]: just making sure this was pushed to ECR

**Review by paulineribeyre — APPROVED** [review]

File: qa-covid19.planx-pla.net/manifest.json (hunk 5 above)
Comment (atharvar28) [reply qa-covid19.planx-pla.net/manifest.json:12]: Yes it is pushed to ECR

**Comment by PlanXCyborg** [comment]
# qa-covid19.planx-pla.net/manifest.json
## Deployment changes
- arborist
  - Needs Fence later than 4.22.2 or 2020.08 (which introduces the new JWT scopes claim). (uc-cdis/arborist#130)
- awshelper
  - The new version of the audit service will require running `gen3 kube-setup-audit-service` and `gen3 kube-setup-fence` again, and updating the audit-service and fence configuration files (uc-cdis/cloud-automation#1603)
- fence
  - By default Fence will allow only visas issued by RAS staging, RAS prod, and itself; if a different set of allowed visa issuers is needed, edit the GA4GH_VISA_ISSUER_ALLOWLIST config value. Otherwise, no action needed. (uc-cdis/fence#933)
  - /link/google will DENY requests with "redirect" param values that aren't in the "allowed redirects" for OAuth2/OIDC Clients in Fence (You have to add any valid redirects to the client before deploying this) (uc-cdis/fence#927)
  - For existing deployments with "ENABLE_AUDIT_LOGS" turned ON, to DISABLE pushing audit logs to an AWS SQS, the configuration "PUSH_AUDIT_LOGS_CONFIG.type" must be changed to "api" (default: "aws_sqs"). If audit-service has never been deployed, no action required. (uc-cdis/fence#923)
  - To ENABLE pushing audit logs to an AWS SQS (REQUIRES audit-service 1.0.0 or more recent), create/update the configuration file's "PUSH_AUDIT_LOGS_CONFIG" field ("type: aws_sqs" (default), and the SQS URL and region can be copied from the audit-service configuration - or run `gen3 sqs info $(gen3 api safe-name audit-sqs)`), see https://github.com/uc-cdis/fence/blob/fec3dd937d68384e86b2ed6f9b53daf0026c2df9/fence/config-default.yaml#L632-L636. Then run `kubectl delete secret fence-config` and `gen3 kube-setup-fence` (uc-cdis/fence#923)
  - This commit includes patches to allow old style refresh tokens and API keys to pass new style validation, while issuing new style refresh tokens and API keys. Since refresh tokens and API keys have a default TTL of 30 days, stay on this tag for 30 days if possible. (uc-cdis/fence#839)
  - Old access tokens will NOT work once this is deployed so active sessions may be temporarily interrupted. (uc-cdis/fence#839)
- indexd
  - When running gen3authz,  backoff package needed separate poetry run pip install (uc-cdis/indexd#304)

**Pull request merged**